fn/goroutine_manager.go

```diff
@@ -8,46 +8,55 @@ import (
 // GoroutineManager is used to launch goroutines until context expires or the
 // manager is stopped. The Stop method blocks until all started goroutines stop.
 type GoroutineManager struct {
-	wg     sync.WaitGroup
-	mu     sync.Mutex
-	ctx    context.Context
-	cancel func()
+	stopped sync.Once
+	wg      sync.WaitGroup
+	mu      sync.Mutex
+	quit    chan struct{}
 }
 
 // NewGoroutineManager constructs and returns a new instance of
 // GoroutineManager.
-func NewGoroutineManager(ctx context.Context) *GoroutineManager {
-	ctx, cancel := context.WithCancel(ctx)
-
+func NewGoroutineManager() *GoroutineManager {
```

Comment:
:+1: 
In the original implementation there were two ways how to stop everything: via ctx and via Stop method. I like that now there is only a single way to do this!

Reply:
yeah: one way to stop the whole manager but then also a context for each `Go` call can be used to stop an individual goroutine